src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        result.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.every((element) => delete newState[element]);
+        result.push({ ...newState });
+        break;
+
+      case 'clear':
+        for (const key in newState) {
+          delete newState[key];
+        }
+
+        result.push({ ...newState });
```

Comment:
You do push for each case. Just move it out of the switch...case and do it after all iterations